data_prep/introspector.py

```diff
@@ -308,9 +337,9 @@ def _group_function_params(param_types: list[str],
 
 
 def populate_benchmarks_using_introspector(project: str, language: str,
-                                           limit: int):
+                                           limit: int, target_oracle: str):
   """Populates benchmark YAML files from the data from FuzzIntrospector."""
-  functions = get_unreached_functions(project)
+  functions = query_introspector_for_targets(project, target_oracle)
```

Comment:
nit: Shall we remove functions `get_unreached_functions()` and `query_introspector_for_unreached_functions()` since anyone will no longer use them after this change?

Reply:
I refined a bit, please check. I kept `get_unreached_functions` because I noticed it had a bit of post-processing in the code which I thought we should keep. That made think in general it's nice to be able to process a bit of the output from the FI APIs so made the oracle dictionary a dictionary to functions rather than APIs, the functions can themselves have the relevant APIs now.

Reply:
Thanks for refining the code.
There appears to be a conflict (after merging #180); could you please resolve that? Much appreciated.

Reply:
Rebasing done

Reply:
Amazing, thanks!
Approved with a final nit comment regarding leading/trailing spaces in the latest commit.